app/models/portfolio.rb

```diff
@@ -24,7 +25,20 @@ def add_portfolio_item(portfolio_item)
   end
 
   def metadata
-    {:user_capabilities => user_capabilities,
-     :shared            => self.access_control_entries.any?}
+    ancillary_metadata.metadata_attributes.merge(:user_capabilities => user_capabilities)
+  end
+
+  private
+
+  def update_ancillary_metadata
+    ancillary_metadata.statistics = statistics_metadata
+  end
+
+  def statistics_metadata
+    {
+      'approval_processes' => tags.where(:name => 'workflows', :namespace => 'approval').count,
+      'portfolio_items'    => portfolio_items.count,
+      'shared_groups'      => access_control_entries.count { |ace| ace.permissions.present? }
```

Comment:
@mkanoor @eclarizio While adding tests to cover unshare I realized there was a bigger issue with `shared_groups` line.

The way the ACEs are working today if you delete the permissions you still have a ACE record but the permissions are empty.  (You can see that we are skipping them during share_info here: https://github.com/RedHatInsights/catalog-api/blob/master/app/services/api/v1x0/catalog/share_info.rb#L22)

So I changed this logic, which seems correct to me, but something else is going on as it is causing numerous test failures.  Was having some difficulty figuring out the root cause last night.

Could really use some help on this one.

Also, wondering if we should consider deleting the ACE if there are no permissions.  Then this call becomes a simple `count` method.

Reply:
@gmcculloug @eclarizio @syncrou 
This was part of the optimizations that we did to have join tables between permission and ACES. So now ACES have a relationship to permissions, the permissions are never deleted you just get an empty array from the relationships call. That permissions empty check was added so that we dont show empty permissions to users after a share was removed. 
https://github.com/RedHatInsights/catalog-api/pull/689